### ISSUE TITLE
[SR-7118] REPL :type lookup of entire Swift module aborts (due to assertion failure)

### ISSUE THREAD

**Issue by benrimmington** [swiftlang/llvm-project#4376]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-7118      |
   |Radar             | rdar://problem/38142245         |
   |Original Reporter | @benrimmington      |
   |Type              | Bug    |
   |Status            | Closed        |
   |Resolution        | Done    |
        
<details>
  <summary>Environment</summary>

-   macOS 10.13.3 (17D102)

-   Xcode 9.2 (9C40b)

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 418fe1c3b7cd947817840c09abbde275

  </details>

**Issue Description:**

The `:type lookup Swift` command in the lldb REPL fails (after printing approximately a thousand lines).

I've tested with the latest [snapshots](https://swift.org/download/#snapshots) dated March 3, 2018.

``` java
$ xcrun --toolchain org.swift.3020180304a swift
Welcome to Apple Swift version 4.2-dev (LLVM c4ec2ab808, Clang af436f313e, Swift 26c08fb346). Type :help for assistance.
  1> :type lookup Swift

Assertion failed: (signature || !origType->hasTypeParameter()), function initSwiftType, file /Users/buildnode/jenkins/workspace/oss-swift-package-osx/swift/include/swift/SIL/AbstractionPattern.h, line 344.
Abort trap: 6
```

``` java
$ xcrun --toolchain org.swift.4120180303a swift
Welcome to Apple Swift version 4.1-dev (LLVM 1c8b50929b, Clang 420ae40df6, Swift b2d7fb958c). Type :help for assistance.
  1> :type lookup Swift

Assertion failed: (signature || !origType->hasTypeParameter()), function initSwiftType, file /Users/buildnode/jenkins/workspace/oss-swift-4.1-package-osx/swift/include/swift/SIL/AbstractionPattern.h, line 344.
Abort trap: 6
```

``` java
$ swift
Welcome to Apple Swift version 4.0.3 (swiftlang-900.0.74.1 clang-900.0.39.2). Type :help for assistance.
  1> :type lookup Swift

Stack dump:
0.  HandleCommand(command = "type lookup Swift")
Segmentation fault: 11
```

The `:print_module Swift` command in the integrated REPL succeeds.

**Comment by belkadan** [comment]
Crashing in the compiler part.

@swift-ci create

**Comment by benrimmington** [comment]
Fixed in Xcode 10.0 (10A255) with Swift 4.2 REPL.
